File: src/main/java/com/viaversion/aas/codec/packet/login/SetCompression.java
package com.viaversion.aas.codec.packet.login;

import com.viaversion.aas.codec.packet.Packet;
import com.viaversion.viaversion.api.type.Type;
import io.netty.buffer.ByteBuf;
import org.jetbrains.annotations.NotNull;

public class SetCompression implements Packet {
	private int threshold;

	public int getThreshold() {
		return threshold;
	}

	public void setThreshold(int threshold) {
		this.threshold = threshold;
	}

	@Override
	public void decode(@NotNull ByteBuf byteBuf, int protocolVersion) throws Exception {
		threshold = Type.VAR_INT.readPrimitive(byteBuf);
	}

	@Override
	public void encode(@NotNull ByteBuf byteBuf, int protocolVersion) throws Exception {
		Type.VAR_INT.writePrimitive(byteBuf, threshold);
	}
}

File: src/main/kotlin/com/viaversion/aas/packet/login/SetCompression.kt
package com.viaversion.aas.packet.login

import com.viaversion.aas.packet.Packet
import com.viaversion.viaversion.api.type.Type
import io.netty.buffer.ByteBuf
import kotlin.properties.Delegates

class SetCompression : Packet {
    var threshold by Delegates.notNull<Int>()
    override fun decode(byteBuf: ByteBuf, protocolVersion: Int) {
        threshold = Type.VAR_INT.readPrimitive(byteBuf)
    }

    override fun encode(byteBuf: ByteBuf, protocolVersion: Int) {
        Type.VAR_INT.writePrimitive(byteBuf, threshold)
    }
}